sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/spanner/SpannerWriteSchemaTransformProvider.java

```diff
@@ -69,6 +74,31 @@ static class SpannerSchemaTransformWrite implements SchemaTransform, Serializabl
       this.configuration = configuration;
     }
 
+    // A generic counter for PCollection of Row. Will be initialized with the given
+    // name argument. Performs element-wise counter of the input PCollection.
+    private static class ElementCounterFn extends DoFn<Row, Row> {
+
+      private Counter spannerGenericElementCounter;
+      private Long elementsInBundle = 0L;
+
+      ElementCounterFn(String name) {
+        this.spannerGenericElementCounter =
```

Comment:
what is this being used for? I ask because counters are approximate in streaming